### DISCUSSION TITLE
Bulk change storage location

### DISCUSSION THREAD

**Discussion by w0hc** [Part-DB/Part-DB-server#486] [Ideas]
I have a ton of parts that are currently stored in Storage Location 1, but I want to move those parts (but not all the parts in Location 1) to Storage Location 4. Can a way be added to select all those parts and move them?

**Comment by isaacolsen94** [comment]
I'm also having this problem. I created a bunch of parts before figuring out storage only to find I can't bulk assign them. This feature would be amazing!

**Comment by dtk1985** [comment]
+1

**Comment by RaptorDE** [comment]
# Moving Components to Another Storage Location with SQL

> [!CAUTION]
> Working directly with SQL can lead to **data loss**. This process is **not supported by the developer** and should only be performed by someone who fully understands the implications.

> [!NOTE]
> When changing the storage location, the filter is set to the attribute **"Favorite"**.

## Steps:

> [!IMPORTANT]
> 1. Create a backup of the database. [See Documentation](https://docs.part-db.de/usage/backup_restore.html)

2. Determine the IDs of the **current** and **target** storage locations from the URL.  
   (e.g., `/en/store_location/40/` → ID = `40`)

3. For all components to be moved, set the attribute **"Favorite"**. 

4. List all **favorited components** ```https://part-db.herokuapp.com/en/parts?part_filter%5Bfavorite%5D%5Bvalue%5D=true&part_filter%5Bsubmit%5D=``` and verify if they are located in the **current** storage location.

5. Insert the IDs of the **current** and **target** storage locations into the SQL query.

6. Execute the **SELECT** SQL query and compare the results with the components that need to be moved.

```sql
-- Declaration and assignment of variables
SET @current_id = 1;
SET @target_id = 70;

SELECT *
FROM part_lots
WHERE id_part IN (
    SELECT id
    FROM parts
    WHERE favorite = 1
)
AND id_store_location = @current_id
-- AND ISNULL(id_store_location) -- Use this variant if the storage is NULL (unknown location)
;
```

7. Execute the UPDATE SQL query and verify the results in the database.

```sql
-- Declaration and assignment of variables
SET @current_id = 1;
SET @target_id = 70;

UPDATE part_lots
SET id_store_location = @target_id
WHERE id_part IN (
    SELECT id
    FROM parts
    WHERE favorite = 1
)
AND id_store_location = @current_id
-- AND ISNULL(id_store_location) -- Use this variant if the storage is NULL (unknown location)
;
```

**Comment by mkne** [comment]
I would also like to have the possibility to change the storage location in bulk like I can change the category for mutiple parts at once. However, I think because you can have multiple storage locations for a part that it is not as easy to implement as for the category.
